cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security;
+
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.twill.LocalLocationFactory;
+import co.cask.cdap.gateway.handlers.InMemoryAuthorizer;
+import co.cask.cdap.internal.test.AppJarHelper;
+import co.cask.cdap.proto.NamespaceMeta;
+import co.cask.cdap.proto.id.InstanceId;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.security.Action;
+import co.cask.cdap.proto.security.Principal;
+import co.cask.cdap.security.spi.authentication.SecurityRequestContext;
+import co.cask.cdap.security.spi.authorization.UnauthorizedException;
+import co.cask.cdap.test.TestBase;
+import co.cask.cdap.test.TestConfiguration;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Files;
+import org.apache.twill.filesystem.Location;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.io.IOException;
+
+/**
+ * Unit tests with authorization enabled.
+ */
+public class AuthorizationTest extends TestBase {
+  private static Location extensionJar;
+  static {
+    LocalLocationFactory locationFactory = new LocalLocationFactory(Files.createTempDir());
+    try {
+      extensionJar = AppJarHelper.createDeploymentJar(locationFactory, InMemoryAuthorizer.class);
+    } catch (IOException e) {
+      throw Throwables.propagate(e);
+    }
+  }
+  private InstanceId instance;
+
+  private final Principal alice = new Principal("alice", Principal.PrincipalType.USER);
+
+  @ClassRule
+  public static final TestConfiguration TEST_CONFIG = new TestConfiguration(
+    Constants.Security.ENABLED, true,
+    Constants.Security.Authorization.ENABLED, true,
+    Constants.Security.Authorization.EXTENSION_JAR_PATH, extensionJar.toURI().getPath()
+  );
+
+  @BeforeClass
+  public static void setup() {
+    SecurityRequestContext.setUserId("alice");
```

Comment:
constant string for 'alice'? since I guess it should be the same used in line 60?